src/gui/gui_utils/netlist.cpp

```diff
@@ -0,0 +1,63 @@
+#include "gui_utils/netlist.h"
+#include "gui/gui_globals.h"
+
+#include "netlist/gate.h"
+#include "netlist/module.h"
+
+// TODO Consider these for moving into the core if they are useful
+
+#include <unordered_set>
+
+namespace gui_utility
+{
+
+    std::shared_ptr<module> common_ancestor_two(std::shared_ptr<module> m1, std::shared_ptr<module> m2)
```

Comment:
That name is a bit weird. @RenWal you try to find a common ancestor for two modules here?

For me, the difference to common_ancestor is not completely clear.

Reply:
Exactly. What both methods do is calculate the [lowest common ancestor](https://en.wikipedia.org/wiki/Lowest_common_ancestor) in the module hierarchy tree, meaning the closest (least hops away) parent module that has a path to both modules.

`common_ancestor_two` finds the LCA for exactly two modules. `common_ancestor` utilizes that method internally and finds the LCA for an arbitrary amount of gates and modules.

Reply:
Well ordinarily this would seem like a textbook example for the use case of function overloading, but in this specific case I think the two should probably just be generalized into a single function that finds the lowest common ancestor for an arbitrary set of modules.

Reply:
We can make these overloaded, which would get rid of the `_two`.

Reply:
@RenWal I think overloading or renaming to `common_ancestor_multimodule` would be an option.